mm2src/mm2_metrics/src/adapt.rs

```diff
@@ -0,0 +1,381 @@
+use std::{collections::HashMap,
+          slice::Iter,
+          sync::{atomic::Ordering, Arc, Mutex}};
+
+use fomat_macros::wite;
+use gstuff::ERRL;
+use itertools::Itertools;
+use metrics::{try_recorder, IntoLabels, KeyName, Label, Recorder, Unit};
+use metrics_core::ScopedString;
+use metrics_exporter_prometheus::formatting::key_to_parts;
+use metrics_util::registry::{GenerationalAtomicStorage, Registry};
+use serde_json::Value;
+use std::fmt::Write;
+
+use crate::{common::log::Tag,
+            new_lib::{MetricType, MetricsJson}};
+
+/// Increment counter if an MmArc is not dropped yet and metrics system is initialized already.
+#[macro_export]
+macro_rules! mm_counter_new {
+    ($metrics:expr, $name:expr, $value:expr) => {{
+            if let Some(recorder) = $crate::new_lib::TryRecorder::try_recorder(&$metrics){
+                 let key_name = $crate::metrics::KeyName::from_const_str(&$name);
+                 let key = $crate::metrics::Key::from_name(key_name);
```

Comment:
These two lines are repeated in every macro

Reply:
Done